src/main/java/com/bravo/user/dao/repository/PaymentRepository.java

```diff
@@ -0,0 +1,15 @@
+package com.bravo.user.dao.repository;
+
+import java.util.List;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import com.bravo.user.dao.model.Payment;
+
+@Repository
+public interface PaymentRepository extends JpaRepository<Payment, String> {
```

Comment:
This should be extending PagingAndSortingRepository. And the findByUserId() method should include a Pageable parameter.

Reply:
Hi Darrin,

Good call on the paging. Yeah at first I had it set up with the paging specifications, but then took it out because it didn't look like the requirements asked for paging. So went ahead and added it back in and updated the unit tests. I added the screenshots in the reply for the other comment. Thanks!

Reply:
Thanks @seancarnahan !! Yes that's exactly what I was looking for. You took it a step further with JpaSpecificationExecutor. The bottom line was to tell the Repository how to page the results (which can be done in several ways). Looks better, thank u.